### DISCUSSION TITLE
Is it possible to retrieve data from an existing redis?

### DISCUSSION THREAD

**Discussion by LucasAndFlores** [OptimalBits/bull#2520] [Q&A]
I tried to find this from the docs and also take a look on issues and didn't find nothing about it. 
I have a Redis that I would like to retrieve all the data inside and use the library to work with it, is it possible? 
I mean, imagine that I have 30 keys inside a Redis and I'd like to work them with the library, how should I proceed?

**Comment by manast** [comment]
I am not sure I understand your question.

**Comment by LucasAndFlores** [comment]
@manast Let's say that I already have a Redis populated with data, like 
```
[
	{name: 'test', email: 'test@test.com.br'},
	{name: 'lucas', email: 'lucas@test.com.br'},
	{name: 'brain', email: 'brain@test.com.br'}
]
```
How should I use bull in this case? Because I would like to process this data which is already inside of this Redis. And today I tried to do something like 
```
const queue = new Bull('testing-queue', {redis: {port: 6379, host: 'redis-local'}})

function main() {
	queue.process((job) => {
		console.log(job.data)
	})
}
main()
```
And didn't work.

**Reply by manast** [reply]
Bull uses its own Redis structures, it cannot use data that you just stored there in arbitrary keys.
